src/main/java/com/bootcamp/demo/Controller/CardController.java

```diff
@@ -0,0 +1,21 @@
+package com.bootcamp.demo.Controller;
+
+import com.google.api.core.ApiFuture;
+import com.google.cloud.firestore.Firestore;
+import com.google.cloud.firestore.WriteResult;
+import com.google.firebase.cloud.FirestoreClient;
+import org.springframework.web.bind.annotation.*;
+
+
+@RestController
+@RequestMapping("/card")
+public class CardController {
+    private Firestore firestoreDB;
+
+    @DeleteMapping(value = "/{cardNumber}")
+    public String removeCard(@PathVariable String cardNumber){
+        this.firestoreDB = FirestoreClient.getFirestore();
```

Comment:
The `CardController` bean should not create / fetch new instances for its dependencies. All dependencies should be injected by Spring through a constructor. The business logic code from this controller should be able to just use those dependencies directly. 
Hint: You could extract all the `Firestore` related "logic" from `FirebaseController` into a new bean (e.g. a DAO / Repository) that would expose an API (methods) for CRUD operations (thus making an abstraction for the actual DB) and this new bean should be used all across the application by being injected (by Spring) as a dependency where it's needed.

Reply:
I've created a FirebaseDao interface to make the CRUD operations abstract having only id and collectionName as parameters for remove and then I created a CardService to call the firebaseDao remove method and now the controller uses the card service instead of instantiating firebasedb as before